File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> h1 = new HashMap<Integer,Integer>();
        for(int i=0;i<nums.length;i++)
        {
            int difference = target - nums[i];
            if (h1.containsKey(difference))
            {
                return new int[] {h1.get(difference),i};
            }
            h1.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();
    const n = nums.length;

    for(let i = 0; i < n; i++){
        const reqNum = target-nums[i];
        if(map.has(reqNum)){
            return [map.get(reqNum),i];
        }else{
            map.set(nums[i],i);
        }
    }

};